### PR TITLE
Add plugin support for quiz module

### PR DESCRIPTION
**Pull request by dwinn**
This implements support for plagiarism in essay questions for the quiz module, ready for when Moodle support plagiarism.

### CODE REVIEW

File: classes/quiz.class.php [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+ * @package   plagiarism_turnitinsim
+ * @copyright 2020 Turnitin
+ * @author    David Winn <dwinn@turnitin.com>
+ * @license   http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+defined('MOODLE_INTERNAL') || die();
+
+/**
+ * Helper class for plagiarism_turnitinsim component in workshops
```
Comment (matdelong) [inline classes/quiz.class.php:29]: This will sound silly... but Does the checkstyle fail if the comment doesn't end in a period?

**Review by matdelong — APPROVED** [review]
Looks good to me from what I can tell 👍

**Review by CharlotteSpinks — APPROVED** [review]

File: classes/quiz.class.php (hunk 1 above)
Comment (CharlotteSpinks) [reply classes/quiz.class.php:29]: I think it does, yes. Good eyes!

Comment (dwinn) [reply classes/quiz.class.php:29]: I think that's only for single line comments. Bigger mistake here is not changing the word "workshops" to "quiz". I'll run it through the checkstyle before I finish to double check.

**Review by jmcgettrick — APPROVED** [review]
LGTM 👍

**Review by jmcgettrick — APPROVED** [review]

**Review by grijeshsaini — APPROVED** [review]

**Comment by dwinn** [comment]
Merging this as has passed QA.

**Pull request merged**